### DISCUSSION TITLE
Disable/restrict self service for some users

### DISCUSSION THREAD

**Discussion by andi0b** [zitadel/zitadel#7361] [Q&A] [answered]
Is it possible to disable or remove some capabilities from the [self service page](https://zitadel.com/docs/concepts/features/selfservice)  of the user? If I got it right every user can log in to an instance they have an account with and edit their profile and manage the authentication methods. This is great, but in many cases you might want to restrict that. 

It would be great if there was some control on what a user can do, this includes for example:
- disallow changing their name
- disallow adding/changing email address/phone number for 2FA
- hide some sections
- block access completely

Those settings need to be applied on a per-user or per-group basis, not generally for the whole instance. 

In many enterprise environments name and contact information is managed centrally and users should not be able to change them themselves. This could enable them to impersonate someone else (eg. change their name to the name of the CEO and then the ceos name might show up as "requester" for a bank transfer of 1M$). I know that there are also other measures that should be taken to prevent this, but still. 

I used keycloak before and there it's possible to do many of those things just via claims. The profile dashboard is a regular oauth application that can be locked down be removing some claims.

**Comment by hifabienne** [comment, marked as answer]
Hei @andi0b 
At the moment it is not possible to configure which parts should be enabled and which not.
We have implemented a default page, that suits most of the use cases.
Our recommendation is to building the profile page directly into your applications, so you are able to customize it for your needs.

We do have some open issues about that topic, but currently no eta on that.

Restrict users form changing user infos: https://github.com/zitadel/zitadel/issues/7077
Prevent users from accessing Console: https://github.com/zitadel/zitadel/issues/7316

**Reply by andi0b** [reply]
Thanks for the link to #7316 didn't find that on my own, it's a viable solution for "block access completely".
